nightfall-client/src/services/transfer.mjs

```diff
@@ -199,9 +204,10 @@ async function transfer(transferParams) {
       .filter(commitment => commitment.preimage.zkpPublicKey.hex(32) === senderZkpPublicKey.hex(32))
       .forEach(commitment => storeCommitment(commitment, senderZkpPrivateKey)); // TODO insertMany
     // mark the old commitments as nullified
-    // oldCommitments.forEach(commitment => markNullified(commitment));
+    oldCommitments.forEach(commitment => markNullified(commitment));
     return { rawTransaction, transaction: optimisticTransferTransaction };
   } catch (err) {
+    oldCommitments.forEach(commitment => clearNullified(commitment));
```

Comment:
- This doesnt look like the right usage for `clearNullified` as that expects a blockNumberL2?
- I **think** this is redundant because `markNullified` is the last call in the `try` block anyways (so we won't have prematurely marked) and i don't see how it would throw an error (i believe it may only do so for upserts)

Reply:
This is based on a change I did for NF2 where the try block also included the Timber call, so it could fail there but (IIRC) the commitment could be marked as `pendingNullification`, so I did a clear to make sure that change was reverted. Obviously here the `clearNullified` would fail so I'll remove it!

Reply:
Yeah good point. There will be some places where the state of the commitment DB will need to be cleaned up now before exiting. Where you have put it should be fine as that will clean it up (as long as it's an idempotent update).